### PR TITLE
Refactor trace analytics in main to merge into observability

### PR DESCRIPTION
**Pull request by joshuali925**
### Description
This PR merges all commits in `main` branch and trace analytics components to observability branch.
**The overall diff in this PR includes everything in the current trace analytics and might be too large to review. It would be easier to look at this diff** https://github.com/opensearch-project/trace-analytics/compare/f714958d936b711acfb79b62645f541df82351d4...1bb58881feed93520a74a0ea61153fa9002f4317, which only includes commits to refactor current trace analytics to observability as a side bar item.

Unit and integration tests will be updated in a separate PR.

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/trace_analytics/components/common/index.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+// export * from './helper_functions';
+// export { SearchBar, SearchBarProps, renderDatePicker } from './search_bar';
+// export { Plt } from './plots/plt';
```
Comment (ps48) [inline public/components/trace_analytics/components/common/index.ts:29]: I think these comments can be removed.

File: public/components/trace_analytics/components/common/index.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+export const RAW_INDEX_NAME = 'otel-v1-apm-span-*';
+export const SERVICE_MAP_INDEX_NAME = 'otel-v1-apm-service-map*';
```
Comment (ps48) [inline public/components/trace_analytics/components/common/index.ts:32]: Indices and route can be moved to https://github.com/opensearch-project/trace-analytics/tree/feature/observability/common/constants as they are used in the server part too. You may create a new trace-analytics file here.

**Review by mengweieric — APPROVED** [review]
LGTM

File: public/components/trace_analytics/components/app.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+              path={['/dashboard', '/']}
+              render={(props) => renderPageWithSidebar(<Dashboard {...commonProps} />, 1)}
+            />
+            <Route
+              exact
+              path="/traces"
+              render={(props) => renderPageWithSidebar(<Traces {...commonProps} />, 2)}
+            />
+            <Route
+              path="/traces/:id+"
```
Comment (ps48) [inline public/components/trace_analytics/components/app.tsx:116]: Have you tried these routes with both '/trace_analytics', '/trace_analytics/home' prefixes? For me it was giving some issue for custom_panel_view page taking in param.

**Review by ps48 — APPROVED** [review]
Minor comments LGTM

File: public/components/trace_analytics/components/app.tsx (hunk 3 above)
Comment (joshuali925) [reply public/components/trace_analytics/components/app.tsx:116]: ah forgot to remove this file, it got moved to `home.tsx` and is not used anymore, thanks for catching this

File: public/components/trace_analytics/components/common/index.ts (hunk 1 above)
Comment (joshuali925) [reply public/components/trace_analytics/components/common/index.ts:29]: will remove

File: public/components/trace_analytics/components/common/index.ts (hunk 2 above)
Comment (joshuali925) [reply public/components/trace_analytics/components/common/index.ts:32]: will do, thanks

**Pull request merged**